pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
@pmorie cautioned me strongly at the outset against modifying the actual emptyDir implementation in k8s, so the plan was to avoid that if at all possible (which it turned out it was), and check in with @pmorie about how we could adapt it to the actual kube API in the future.

However @pmorie later softened on that stance and if we're sure, I can re-do the plugin code and push it into a patch we carry on top of kube.

However to make the case for the current implementation, it offers a lower maintenance burden and will hopefully one day move to a permanent API location. It relies on volume plugin builder's NewBuilder method, and the volume plugin's SetUp and SetUpAt. I don't know how stable these methods are, I would hope somewhat, but indeed they could change upstream. 

Please let me know if we're sure and I'll get to work on moving everything into kube.